### PR TITLE
add instrumentation hook

### PR DESCRIPTION
**Pull request by erwinmombay**

### CODE REVIEW

**Comment by ampsauce** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 4d6633c00d1ec074c05ac331ec9152b4ca13e3f4
Build details: https://travis-ci.org/ampsauce/amphtml/builds/87092655

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by ampsauce** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 691a3178b2b03b856825f62fa0e8b4c496212af5
Build details: https://travis-ci.org/ampsauce/amphtml/builds/87092544

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by erwinmombay** [comment]
@ampsauce retry

**Comment by ampsauce** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 4d6633c00d1ec074c05ac331ec9152b4ca13e3f4
Build details: https://travis-ci.org/ampsauce/amphtml/builds/87094293

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by erwinmombay** [comment]
@ampsauce retry

**Comment by ampsauce** [comment]
**Tests failed.** Automated cross-browser testing via Sauce Labs and Travis CI shows that the JavaScript changes in this pull request are: **BUSTED**

Commit: 4d6633c00d1ec074c05ac331ec9152b4ca13e3f4
Build details: https://travis-ci.org/ampsauce/amphtml/builds/87118261

(*Please note that this is a [fully automated](https://github.com/twbs/savage) comment.*)

**Comment by erwinmombay** [comment]
@ampsauce retry

**Comment by erwinmombay** [comment]
@cramforce can you give this a first pass review? thanks!

File: src/performance.js [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+      this.tick_(tickEvent.label, tickEvent.opt_from, tickEvent.opt_value);
+    });
+    this.events_.length = 0;
+  }
+
+  /**
+   * @param {funtion(string,?string=,number=)} tick
+   * @export
+   */
+  setPerformanceFunction(tick) {
```
Comment (cramforce) [inline src/performance.js]: How about `setTickFunction`?

File: src/performance.js [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+    }
+
+    this.events_.forEach((tickEvent) => {
+      this.tick_(tickEvent.label, tickEvent.opt_from, tickEvent.opt_value);
+    });
+    this.events_.length = 0;
+  }
+
+  /**
+   * @param {funtion(string,?string=,number=)} tick
```
Comment (cramforce) [inline src/performance.js]: Document required interface

File: src/performance.js (hunk 1 above)
Comment (erwinmombay) [inline src/performance.js]: :+1: sgtm

File: src/performance.js [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {getService} from './service';
+import {timer} from './timer';
+
+
+/** @const {number} */
+const QUEUE_LIMIT_ = 100;
```
Comment (cramforce) [inline src/performance.js]: Lets go down a bit to 50.
Please document.

File: src/amp.js [hunk 4]
```diff
@@ -43,6 +44,7 @@ try {
     try {
```
Comment (cramforce) [inline src/amp.js]: Install performanceFor before installStyles and tick there and at the end of this callback as our first 2 ticks.

**Comment by cramforce** [comment]
Looks great!

File: src/amp.js (hunk 4 above)
Comment (erwinmombay) [inline src/amp.js]: done.

File: src/performance.js (hunk 1 above)
Comment (erwinmombay) [inline src/performance.js]: done.

File: src/performance.js (hunk 2 above)
Comment (erwinmombay) [inline src/performance.js]: done.

File: src/performance.js (hunk 3 above)
Comment (erwinmombay) [inline src/performance.js]: done.

**Comment by erwinmombay** [comment]
@cramforce PTAL

**Comment by erwinmombay** [comment]
@cramforce ping pong

File: src/amp.js [hunk 5]
```diff
@@ -39,6 +40,9 @@ import {maybeValidate} from './validator-integration';
 try {
   // Should happen first.
   installErrorReporting(window);  // Also calls makeBodyVisible on errors.
+  let perf = performanceFor(window);
+
+  perf.tick('installStyles');
```
Comment (cramforce) [inline src/amp.js]: Its pretty common to give these super short names like "is". Maybe create an .md file with a table?

**Comment by cramforce** [comment]
LGTM with one comment.

File: src/amp.js (hunk 5 above)
Comment (erwinmombay) [inline src/amp.js]: got it. would the end one be `endIs` ?

Comment (cramforce) [inline src/amp.js]: e_is :)

Comment (erwinmombay) [inline src/amp.js]: done. added TICKEVENTS.md at root dir. will merge tomorrow morning if you have no further comments with the latest changes.

**Pull request merged**